### DISCUSSION TITLE
Diving into Amazon SP-API Hot Topics! 🔥👋

### DISCUSSION THREAD

**Discussion by weilinggu** [amzn/selling-partner-api-models#4270] [Announcements]
<h1 align="center">🔥 Amazon SP-API Hot Topics! 🔥</h1>

![logo](https://github.com/user-attachments/assets/b8d27ec5-f5b3-4141-960c-1a9434cff46d)

As the [Amazon Selling Partner API (SP-API)](https://developer-docs.amazon.com/sp-api/) continues to evolve, the team behind this powerful set of tools has been proactively addressing some of the most complex, mission-critical topics that developers face. The team is eager to share their expertise and unique perspectives on issues with significant implications for how businesses leverage the Amazon Marketplace through a series of insightful **_“SP-API Hot Topics”_** articles.

By engaging the community on these important issues, we can develop a deeper understanding of how the SP-API and adjacent technologies are shaping the future of online selling. 🤝

> [!NOTE]  
> Amazon SP-API Hot Topics reside in <a href="https://github.com/amzn/selling-partner-api-samples" target="_blank">Amazon Selling Partner API Samples repository</a>. By clicking the link of the topics, you will be taken to Amazon Selling Partner API Samples repository.

## 📑 Explore Topics
| Use Cases | Description |
|---|---|
| [⚙️ Start here: SP-API Application Setup](#start-here-sp-api-application-setup) | Set up your SP-API application, implement authorization, and handle errors |
| [🏅 Best Practices & SP-API Well-Architected Solutions](#sp-api-best-practices--well-architected-solutions) | Build resilient, high-performing, and scalable SP-API applications |
| [🛒 Orders Management](#orders-management) | Process orders, track transactions, and communicate with buyers |
| [💰 Pricing, Fees & Promotions](#pricing-fees--promotions) | Set competitive prices, estimate fees, and run deals or coupons |
| [📦 Product & Listings Management](#product--listings-management) | Create, update, enrich, and manage product listings on Amazon |
| [🚚Shipping & Fulfillment](#shipping--and-fulfillment) | Ship orders & Items through FBA and MFN. |
| [📊 Reporting & Analytics](#reporting--analytics) | Pull reports, query data, and monitor sales and feedback metrics |

<a id="start-here-sp-api-application-setup"></a>
## ⚙️ Start here: SP-API Application Setup
- ### [**Discovering the Selling Partner API Postman Workspace 🔎**](https://github.com/amzn/selling-partner-api-samples/discussions/210)
  This guide introduces the Selling Partner API official Postman workspace, how to explore and use the available collections, flows and resources to simplify SP-API integration journeys.
  _- Dec 05, 2025_
- ### [**Workshop Introduction: Hands-on experiments for SP-API MCP with Amazon Q Developer**](https://github.com/amzn/selling-partner-api-samples/discussions/202)
  In this comprehensive guide, we'll explore how to combine Amazon Q Developer with the Amazon Selling Partner API using MCP servers, creating a powerful natural language interface for seller operations that lets you query business data through conversation.
  _- Nov 28, 2024_
- ### [**Getting Started with SP-API Error Handling 🎬🛠️**](https://github.com/amzn/selling-partner-api-samples/discussions/186)
  This guide will explore error handling, common errors and important terminology, tools and strategies for error handling.
  _- Nov 6, 2025_
- ### [**SP-API Authorization Lifecycle Management: A Comprehensive Guide to Secure Integration 🔐🚀**](https://github.com/amzn/selling-partner-api-samples/discussions/112)
  This comprehensive guide explores the various aspects of SP-API authorization lifecycle management, from initial setup to ongoing maintenance.
  _- May 7, 2025_

<a id="sp-api-best-practices--well-architected-solutions"></a>
## 🏅 SP-API Best Practices & Well-Architected Solutions
- ### [**Out now: Big Updates to Sample AI Sandbox 🤖🚀**](https://github.com/amzn/selling-partner-api-samples/discussions/532)
  This guide discusses updates made to the Sample AI Sandbox that helps you verify your SP-API integration before going to production.
  _- September 3, 2026_
- ### [**Now Available: Local AI Sandbox for SP-API 🤖🏗️**](https://github.com/amzn/selling-partner-api-samples/discussions/395)
  This guide introduces local AI Sandbox for SP-API, a open-source, local test environment that helps you verify your SP-API integration.
  _- June 9, 2026_
- ### [**Now Available: AI-Assisted SP-API Development: Preventing Hallucinations with Prompt Engineering 🤖**](https://github.com/amzn/selling-partner-api-samples/discussions/435)
  This is a Developer's Guide to AI-Assisted SP-API Development: Preventing Hallucinations with Prompt Engineering
  _- June 5, 2026_
- ### [**Now Available: Local MCP for SP-API — AI-powered Selling Partner API development 🚀📦**](https://github.com/amzn/selling-partner-api-samples/discussions/382)
  This guide introduces local MCP for SP-API, a consolidated npm package that equips AI coding assistants with authoritative SP-API knowledge, actionable developer tools, and workflow automation.
  _- May 28, 2026_
- ### [**🚨 Mastering SP-API 4xx Error Handling 📘🚀**](https://github.com/amzn/selling-partner-api-samples/discussions/195)
  This guide explains the most common SP-API 4xx errors (400, 403, 404, and 429), why they occur and how to fix them. It also provides API-specific troubleshooting guidance to reduce failed calls, avoid throttling, lower API call volume and ultimately reduce operational costs.
  _- Nov 21, 2025_
- ### [**How to Reduce Your SP-API Call Volume 🚀 📉**](https://github.com/amzn/selling-partner-api-samples/discussions/187)
  This guide offers a comprehensive strategy for reducing Amazon Selling Partner API call volume by leveraging error management, event-driven design, and batch/bulk data retrieval techniques to optimize API interactions and prepare for future API usage fees.
  _- Nov 11, 2025_
- ### [**Estimating AWS costs for SP-API notifications 💰📊**](https://github.com/amzn/selling-partner-api-samples/discussions/115)
  Learn how to estimate AWS service costs when implementing SP-API Notifications through SQS and EventBridge workflows with pricing examples for different business scales.
  _- May 12, 2025_
- ### [**Preparing your SP-API Application for Peak Shopping Events 📊🛍️🚀**](https://github.com/amzn/selling-partner-api-samples/discussions/102)
  Guidance and recommendations to ensure your SP-API application is primed for peak shopping events on [Amazon.com](http://amazon.com/) such as prime days, T5 (short for "Turkey 5") and holiday season.
  _- Nov 19, 2024_

<a id="orders-management"></a>
## 🛒 Orders Management
- ### [**Orders API v2026-01-01: Breaking Changes, Enhanced Features & Call Optimization Strategies 🚀**](https://github.com/amzn/selling-partner-api-samples/discussions/254)
  This guide introduces the breaking changes for new Orders API, as well as the benefits you can get after migrating to Orders API v2026-01-01.
  _- Jan 29, 2026_
- ### [**Migrate to Orders API v2026-01-01: Interactive Developer Guide 🚀**](https://github.com/amzn/selling-partner-api-samples/discussions/253)
  This guide walks you through the migration step-by-step with ready-to-run Postman requests at each stage.
  _- Jan 29, 2026_
- ### [**📣 Orders API Guide: How to get latest orders update in an efficient way**](https://github.com/amzn/selling-partner-api-samples/discussions/104)
  Guidance and best practices of using Orders API to get latest order updates with scalability and efficiency.
  _- Jan 14, 2025_
- ### [**SP-API Practical Guide: Efficient Use of Order and Inventory Operation 💡**](https://github.com/amzn/selling-partner-api-samples/discussions/215)
  This guide shows how to optimize SP-API calls for order processing and inventory management. It focuses on using notifications and reports to reduce API costs while meeting business needs.
  _- Dec 09, 2025_

<a id="pricing-fees--promotions"></a>
## 💰 Pricing, Fees & Promotions
- ### [Competing for Featured Offer Using Expected Price (FOEP) 🏷️💰🎯 ](https://github.com/amzn/selling-partner-api-samples/discussions/522)
   This guide explains how to build a repricing solution that responds to market changes by leveraging FOEP and push notifications to unlock a competitive advantage for Featured Offer placement.
 _- August 30, 2026_
- ### [**Optimizing your SP-API Pricing Integration 🎯⚡**](https://github.com/amzn/selling-partner-api-samples/discussions/209)
  This guide covers best practices for optimizing your SP-API pricing integration through efficient call patterns, batch operations, and smart caching strategies to reduce costs and improve performance.
  _- Dec 04, 2025_
- ### [**Using Pricing APIs and Notifications to improve Price Competitiveness 💰💸🏆**](https://github.com/amzn/selling-partner-api-samples/discussions/100)
  Discuss how you can leverage the Selling Partner API's pricing features to stay competitive and win the Buy Box.
  _- Oct 28, 2024_
- ### [**How to Integrate Location-Based Pricing Using Amazon SP-API 🌎💰🔄**](https://github.com/amzn/selling-partner-api-samples/discussions/106)
  A comprehensive guide on implementing location-based pricing strategies using Amazon's SP-API by leveraging Glance View Weight Percentage (GVWP), price thresholds, and real-time notifications.
  _- Apr 11, 2025_

<a id="product--listings-management"></a>
## 📦 Product & Listings Management
- ### [**Practical Guide: Increase your Listings Feed success rate 📈🚀**](https://github.com/amzn/selling-partner-api-samples/discussions/546)
  This guide shares ten practical advices to increase the success rate for listings feed submissions.
  _- September 25, 2026_
- ### [**Managing Handling Time on Amazon ⌛🚚**](https://github.com/amzn/selling-partner-api-samples/discussions/124)
  This guide explains what handling time is, why it matters for Amazon sellers, and outlines various methods and key considerations for managing it—ensuring accurate delivery promises, reliable order fulfillment, and a stronger competitive edge on Amazon.
  _- May 26, 2025_
- ### [**Navigating Listings API Rate Limits: 3 Optimization Strategies for Speedy Listings Management 🚀📊🔍**](https://github.com/amzn/selling-partner-api-samples/discussions/109)
  This guide provides three strategic optimization techniques for navigating Amazon's Selling Partner API Listings rate limits, helping solution providers and seller partners accelerate listings processing while maintaining efficient and quality API submissions.
  _- May 6, 2025_
- ### [**⚙️📚 Automating GPSR Compliance with SP-APIs: Everything You Need to Know**](https://github.com/amzn/selling-partner-api-samples/discussions/101)
  Learn how to streamline GPSR compliance by efficiently submitting product safety documentation using SP-APIs.
  _- Feb 24, 2025_

<a id="shipping--and-fulfillment"></a>
## 🚚 Shipping & Fulfillment
- ### [Handling EU Customs Reform Carrier Validation for FBM Shipments 🇪🇺📦🚢 ](https://github.com/amzn/selling-partner-api-samples/discussions/434)
  This guide is our opinionated take on how integrators should handle this change. Every integration is different, your architecture, UX, and seller workflows will determine the exact implementation.
  _- June 30, 2026_

<a id="reporting--analytics"></a>
## 📊 Reporting & Analytics
- ### [**Introducing the Amazon Data Kiosk MCP Server: Supercharge Your Amazon Analytics with AI 📊🤖🚀**](https://github.com/amzn/selling-partner-api-samples/discussions/131)
  This game-changing tool transforms how sellers and vendors interact with their Amazon business data through the power of AI and natural language processing!
  _- May 15, 2025_

## Here are some of the things you can do:

-  Seek solutions and share Hot Topics with your peers. 📣
- Look for programming help from other members of the Amazon SP-API community. 👨‍💻
- Share what topics are most significant to you, and you would like to see next. 📝
-  Learn how to get the most out of SP-API. 🎯
- Ask questions, but answer questions, too. Share your knowledge! ✨
- Upvote your favorite Hot Topics. This will help other people find the solution. 👍
 - Be respectful of others, and report any posts or comments you see that are spam, derogatory, or otherwise against the rules. 🚫

## Your opinion matters! Let us know what topics do you want to see next.

- Streamline fulfillment process 🚚     
- Best practice in managing your product catalog 📚
- Keeping your product price competitive 💰
- Driving insights for your products using analytics via GraphQL 📊
- And more 🔍

Jump in, share your thoughts, and let's learn from each other! 💬 Remember to be respectful, constructive, and have fun! 🎉

> [!IMPORTANT]  
> ⚠️ For support inquiries, please reach out to support directly. ⚠️

![](https://d1ev96nfd4idq8.cloudfront.net/diving-into-amazon-sp-api-hot-topics)
![](https://drei6xcvcxfyd.cloudfront.net/diving-into-amazon-sp-api-hot-topics)

**Comment by weilinggu** [comment]
NEW HOT TOPIC  ✨

[Preparing your SP-API Application for Peak Shopping Events 📊🛍️🚀](https://github.com/amzn/selling-partner-api-samples/discussions/102)

🔥 This hot topic will help ensure your Amazon Selling Partner API (SP-API) application is ready to handle high traffic and surges during peak shopping events like Prime Day and the holiday season. 

This guide provides insights and strategies to:
- Identify bottlenecks
- Optimize API usage
- Retry techniques
- Proactively manage your application's performance and reliability

Be sure to checkout the new hot topic! 🆕🔥 Comment in the [hot topic](https://github.com/amzn/selling-partner-api-samples/discussions/102) to let us know what do you think 💭💬

**Comment by Tong8s** [comment]
NEW HOT TOPIC ✨

[📣 Orders API Guide: How to get latest orders update in an efficient way](https://github.com/amzn/selling-partner-api-samples/discussions/104)

🔥 This hot topic will discuss how to use Orders APIs to get latest order updates in scalable and efficient way.

This guide talks about 4 ways of retrieving latest orders update and the applied scenarios for each
* `GetOrders` + `GetOrderItems` API Call
*  `ORDER_CHANGE` notification
*  Order Reports
*  Combine and Conquer

Be sure to checkout the new hot topic! 🆕🔥 Comment in the [hot topic](https://github.com/amzn/selling-partner-api-samples/discussions/104) to let us know what do you think 💭💬

**Comment by ritikacd** [comment]
NEW HOT TOPIC ✨

[⚙️📚 Automating GPSR Compliance with SP-APIs: Everything You Need to Know](https://github.com/amzn/selling-partner-api-samples/discussions/101)

🔥 This hot topic explores how SP-APIs can simplify GPSR compliance by enabling efficient submission of required safety attributes and documentation.

This guide covers: 
* Understanding GPSR requirement for product listings
* Required actions by Selling Partners and Developers
* Submitting data using various API and feed types 

Be sure to checkout the new hot topic! 🆕🔥 Comment in the [hot topic](https://github.com/amzn/selling-partner-api-samples/discussions/101) to let us know what do you think 💭💬

**Comment by leonardonakagawa** [comment]
NEW HOT TOPIC ✨

[How to Integrate Location-Based Pricing Using Amazon SP-API 🌎💰🔄](https://github.com/amzn/selling-partner-api-samples/discussions/106)

🔥 Check how SP-APIs can help sellers implement location-based pricing strategies using real-time data and notifications.

This guide covers:

* Best practices for market analysis and competitive positioning
* Understanding GVWP (Glance View Weight Percentage) for location analysis
* Setting up location-specific price targeting
* Leveraging Price Thresholds for Featured Offer optimization
* Implementing real-time pricing updates with ANY_OFFER_CHANGED notifications



Be sure to checkout the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by weilinggu** [comment]
NEW HOT TOPIC ✨

[Navigating Listings API Rate Limits: 3 Optimization Strategies for Speedy Listings Management 🚀📊🔍](https://github.com/amzn/selling-partner-api-samples/discussions/109)

Navigating Amazon's Selling Partner API (SP-API) Listings API demands strategic expertise to manage rate limits and optimize listings processing. 

This guide unveils three powerful strategies that enable solution providers and seller partners to: 
- Understand listings submission rate limit and processing pipeline
- Optimize listings update and creation processing speed
- Maximize API submission efficiency

Be sure to checkout the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by hichenai** [comment]
NEW HOT TOPIC ✨

[SP-API Authorization Lifecycle Management: A Comprehensive Guide to Secure Integration 🔐🚀
](https://github.com/amzn/selling-partner-api-samples/discussions/112)
Managing authorization for Amazon's Selling Partner API (SP-API) requires a strategic approach to handle multiple authorization flows, credential rotation, and security requirements effectively.

This comprehensive guide covers key areas that enable developers and organizations to:
- Implement robust authorization lifecycle management systems
- Handle multiple authorization flows (Website, App Store, and Self-Authorization)
- Optimize security and credential management
- Build scalable serverless implementations using AWS services

Key highlights include:
1. Dual lifecycle management of app client credentials and tenant authorizations
2. Implementation strategies for different authorization flows
3. Grantless operations for infrastructure management
4. Secure credential handling and rotation
5. Serverless architecture using AWS services for scalable solutions

Check out this essential guide for building reliable SP-API integrations! 🆕🔥 Share your thoughts and experiences in the comments below 💭💬

**Comment by leonardonakagawa** [comment]
NEW HOT TOPIC ✨

[Estimating AWS costs for SP-API notifications 💰📊 ](https://github.com/amzn/selling-partner-api-samples/discussions/115)

Estimating AWS service costs when implementing SP-API Notifications requires expertise in AWS pricing models for services like SQS and EventBridge.

This guide covers detailed information about:

- Overview of SQS and EventBridge
- Pricing model for both services
- Examples to estimate AWS cost for different business scales.

Be sure to checkout the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by hichenai** [comment]
NEW HOT TOPIC ✨

[Introducing the Amazon Data Kiosk MCP Server: Supercharge Your Amazon Analytics with AI](https://github.com/amzn/selling-partner-api-samples/discussions/131)📊🤖🚀

🔥 This game-changing tool transforms how sellers and vendors interact with their Amazon business data through the power of AI and natural language processing!

Key highlights include:

✅ Natural language queries - no coding needed!
✅ Real-time sales, traffic, and financial insights
✅ Custom reports and visualizations
✅ Actionable AI-driven recommendations
✅ Seamless integration with Amazon Selling Partner API

Perfect for:
📈 Data-driven sellers
🏪 Vendors seeking deeper insights
💻 Developers looking to prototype quickly
🔍 Anyone wanting to unlock the full potential of their Amazon business data

Setup is quick and straightforward:

Clone the repository
Install dependencies
Configure your MCP client
Start analyzing your data with AI!
Be sure to checkout this innovative solution! 🆕🔥 Comment below to share your experience or ask questions about implementation 💭💬

**Reply by pandjtrading** [reply]
The link returns a 404

**Comment by ritikacd** [comment]
NEW HOT TOPIC ✨

[Managing Handling Time on Amazon](https://github.com/amzn/selling-partner-api-samples/discussions/124)⌛🚚

In this guide, we'll dive deep into Amazon's three different methods for managing handling time: Default Account Settings, SKU-Specific Controls, and Automated Handling Time (AHT). You'll learn how each approach works, when to use them, and how to implement them through both Seller Central and SP-API integrations.

Key Topics:
* Understanding handling time and its impact on your business
* Setting up and managing default handling times
* Implementing SKU-specific handling times for greater control
* Leveraging Automated Handling Time (AHT) for your top 30K SKUs
* Best Practices & Common Pitfalls to avoid

Be sure to checkout the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by mafge** [comment]
NEW HOT TOPIC ✨

[Getting Started with SP-API Error Handling 🎬🛠️](https://github.com/amzn/selling-partner-api-samples/discussions/186)

In this guide, we will have a look at how to tackle error handling when using Selling Partner API by Amazon.

This guide covers detailed information about:

* How you can use Amazon Solution Provider Portal to gain insights
* Error types you might encounter when integrating Selling Partner API
* How to leverage automation to implement logging, monitoring and alerting

Be sure to checkout the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by MeenaAmz** [comment]
NEW HOT TOPIC ✨

[How to Reduce Your SP-API Call Volume 🚀 📉 
](https://github.com/amzn/selling-partner-api-samples/discussions/187)

In this guide, we will provide a strategic approach to reducing Amazon Selling Partner API (SP-API) call volume by analyzing API usage, implementing targeted optimization techniques, and combining notifications with strategic data retrieval.

This guide covers detailed information about:

- Handling high error rates
- Implementing event-driven architecture using SP-API notifications
- Using batch/bulk operations to reduce API call volume
- Optimization checklist with next steps

Be sure to checkout the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by puppsupr** [comment]
NEW HOT TOPIC ✨

[Mastering SP-API 4xx Error Handling](https://github.com/amzn/selling-partner-api-samples/discussions/195) 🚨📘🚀

🔥 In this Hot Topic, we take a practical look at the most common Amazon Selling Partner API (SP-API) 4xx errors, why they occur and how to fix them. This guide builds on [_Part 1_](https://github.com/amzn/selling-partner-api-samples/discussions/187) and [_Part 2_](https://github.com/amzn/selling-partner-api-samples/discussions/186) to help you reduce failed calls, avoid throttling and lower API call volume and costs.

This guide covers:

-   Comprehensive troubleshooting for 400,403,404 and 429 errors
-   API-specific checks for common error-prone APIs like Listings, Catalog, Messaging, Reports & more
-   Strategies to reduce unnecessary API retries and call volume
-   Best practices to enhance reliability and reduce avoidable failures

Be sure to checkout the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by leonardonakagawa** [comment]
NEW HOT TOPIC ✨

[Optimizing your SP-API Pricing Integration 🎯⚡](https://github.com/amzn/selling-partner-api-samples/discussions/209)

This guide covers best practices for optimizing your SP-API pricing integration through efficient call patterns, batch operations, and smart caching strategies to reduce costs and improve performance. 

Be sure to check out the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by hichenai** [comment]
NEW HOT TOPIC ✨

[Bridging AI Assistants and Business APIs: Amazon Q Developer + SP-API with MCP 🤖🔗🚀](https://github.com/amzn/selling-partner-api-samples/discussions/202)

🔥 This comprehensive workshop shows how to combine Amazon Q Developer with the Selling Partner API using Model Context Protocol (MCP) servers, creating a powerful natural language interface for seller operations.

This guide covers:
• Building MCP server bridges between AI assistants and SP-API
• Natural language interactions - ask "Show me my sales data from last month" instead of constructing API requests
• Two specialized servers: SP-API MCP Server for direct API access and Data Kiosk MCP Server for analytics
• Real-world use cases for sellers, developers, and operations teams
• Complete workshop architecture with IAM Identity Center, Amazon Q Developer Pro, and mock API environment

Workshop Details:
• Duration: 60-90 minutes
• Level: Intermediate
• Free tier eligible (Amazon Q Developer Pro subscription required)

Be sure to checkout the new hot topic! 🆕🔥 [Access the hands-on workshop now →](https://catalog.us-east-1.prod.workshops.aws/workshops/339c1a60-098c-4b83-ad61-2e69fcc547ad) Comment in the hot topic to let us know what do you think 💭💬

**Comment by jawheral** [comment]
NEW HOT TOPIC ✨

[Discovering the Selling Partner API Postman Workspace 🔎](https://github.com/amzn/selling-partner-api-samples/discussions/210)

Get to know the Selling Partner API official Postman workspace. Dive into recipes and use-case focused collections and flows that help you move faster in your SP-API integration journey!

Be sure to check out the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by chrisskhoury** [comment]
NEW HOT TOPIC ✨

[Migrate to Orders API v2026-01-01: Interactive Developer Guide 🚀📦💻](https://github.com/amzn/selling-partner-api-samples/discussions/253)

🔥 In this Hot Topic, we walk you through migrating from Orders API v0 to v2026-01-01 with a hands-on, interactive approach using Postman. This isn't just documentation—it's a step-by-step exploration where you can run real API calls and see the improvements firsthand. Achieve up to 65% reduction in API calls while simplifying your code and eliminating RDT complexity.

This guide covers:

- Step-by-step Postman collection – Fork and follow along with pre-configured requests
- API consolidation in action – See how 5 v0 calls become 1 v2026-01-01 call
- Mastering the includedData parameter – Request only what you need for each use case
- Real migration patterns – Daily syncs, order details, fulfillment queues with before/after examples
- Complete field mappings – Navigate PascalCase→camelCase, new status values, and breaking changes
- Cost savings calculator – Understand your potential API call reduction

Be sure to check out the new hot topic! 🆕🔥 Comment in the hot topic to let us know what you think and share your migration experiences 💭💬

**Comment by Tong8s** [comment]
NEW HOT TOPIC ✨

[Now Available: Local MCP for SP-API — AI-powered Selling Partner API development 🚀📦](https://github.com/amzn/selling-partner-api-samples/discussions/382)

🔥 This hot topic introduces local MCP for SP-API, a consolidated npm package that equips AI coding assistants with authoritative SP-API knowledge, actionable developer tools, and workflow automation.


Be sure to checkout the new hot topic! 🆕🔥 Comment in the [hot topic](https://github.com/amzn/selling-partner-api-samples/discussions/382) to let us know what do you think 💭💬

**Comment by mafge** [comment]
NEW HOT TOPIC ✨

[Now Available: Local AI Sandbox for SP-API 🤖🏗️ ](https://github.com/amzn/selling-partner-api-samples/discussions/395)

We're excited to announce the official release of local AI Sandbox for SP-API 🥳🎉 — a open-source, local test environment that helps you verify your SP-API integration.

### Key highlights:
**Request Validation** —Verifies your calls to SP-API and returns comprehensive error messages in case of invalid requests
**Test Complete Workflows** — Allows testing of complete workflows and cross-API use cases
**Dynamic Test Data Generation** — Generates use case specific test data to ensure coverage of happy path and corner cases

Recommended for all Solution Providers building SP-API integrations.

We are excited to see how you are going to use our local AI Sandbox for SP-API. Happy testing! 🔥🚀
Comment in the hot topic to let us know your thoughts 💭💬

**Comment by chrisskhoury** [comment]
NEW HOT TOPIC ✨

[Preparing for the EU Customs Reform: Carrier Compliance for SP-API Developers 📦🇪🇺🚀](https://github.com/amzn/selling-partner-api-samples/discussions/434)

🔥 This hot topic helps you prepare your SP-API integration for the EU Customs Reform regulations taking effect 1 July 2026.

This guide covers:
- Which carriers are approved and their exact CarrierCode values
- How to validate shipments via confirmShipment and POST_ORDER_FULFILLMENT_DATA feed
- Carriers under negotiation (temporary approval through 15 July 2026)
- What happens if you use a non-approved carrier

Be sure to checkout the new hot topic! 🆕🔥 Comment in the hot topic to let us know what you think 💭

**Comment by JingyiMJoy** [comment]
NEW HOT TOPIC ✨
[A Developer's Guide to AI-Assisted SP-API Development: Preventing Hallucinations with Prompt Engineering 🤖
](https://github.com/amzn/selling-partner-api-samples/discussions/435)

🔥 This Hot Topic shows how prompt engineering can improve both AI search and AI coding for SP-API — by grounding responses and generated code in official documentation.

This guide covers:

- Two scenarios: grounded prompts for AI search and structured prompts  for AI coding
- A self-verifying search prompt template + an anchor-URL quick reference for accurate, citable answers
- A universal code generation template (role + context + instructions + example) that makes output more reliable
- Before/after demos showing the hallucination problem and the prompts engineering fix side by side
- What's next: how MCP and Agent Skills could automate grounding for SP-API

Be sure to checkout the new hot topic! 🆕🔥Comment in the hot topic to let us know what you think 💬

**Comment by MeenaAmz** [comment]
NEW HOT TOPIC ✨

[Competing for Featured Offer Using Expected Price (FOEP) 🏷️💰🎯 ](https://github.com/amzn/selling-partner-api-samples/discussions/522)

In this Hot Topic, we explain how to integrate getFeaturedOfferExpectedPriceBatch into a repricing workflow using notifications as triggers. This guide helps developers build responsive pricing solutions that target Featured Offer placement.

Key takeaways:

- Notification-Driven Repricing — Use ANY_OFFER_CHANGED and PRICING_HEALTH as triggers instead of polling FOEP excessively.

- Smart Filtering — Filter by OfferChangeType to reduce unnecessary API calls and focus on competitive landscape shifts.

- Flexible Integration Patterns — Choose from notification-based, periodic polling, or getCompetitiveSummary-combined approaches based on your infrastructure.

Recommended for Solution Providers and Sellers building or enhancing repricing solutions that target Featured Offer placement.

Be sure to check out the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by mafge** [comment]
NEW HOT TOPIC ✨

[Out now: Big Updates to Sample AI Sandbox 🤖🚀](https://github.com/amzn/selling-partner-api-samples/discussions/532)

In this Hot Topic, we are going through the most important changes that we have made to Sample AI Sandbox.

Key updates:

* **Architectural changes**: Reduced latency, no inference cost
* **Additional API Support**: Data Kiosk API, Notifications API
* **Guided Scenarios**: Pre-seeded, runnable SP-API journeys
* **UI Enhancements**: Visual improvements, new features added
* **Vendor Mode**: Test integrations specifically built for vendors

Be sure to check out the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬

**Comment by mafge** [comment]
NEW HOT TOPIC ✨

[Practical Guide: Increase your Listings Feed success rate 📈🚀](https://github.com/amzn/selling-partner-api-samples/discussions/546)

In this Hot Topic, we are going to give ten practical advices on how to increase the success rate for listings feed submissions, including:
* Payload prevalidation
* Optimal feed size ad structure
* Using the right operation type

Be sure to check out the new hot topic! 🆕🔥 Comment in the hot topic to let us know what do you think 💭💬
